README.rst

```diff
@@ -100,16 +100,16 @@ For example:
 * Rewrite ``dict([(1, 2)])`` or ``dict(((1, 2),))`` as ``{1: 2}``
 * Rewrite ``dict([])`` as ``{}``
 
-C407: Unnecessary list comprehension - '<builtin>' can take a generator
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+C407: Unnecessary (dict/list/set) comprehension - '<builtin>' can take a generator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 It's unnecessary to pass a list comprehension to some builtins that can take
 generators instead. For example:
 
-* Rewrite ``sum([x ** 2 for x in range(10)])`` as
+* Rewrite ``sum({x ** 2 for x in range(10)})`` as
```

Comment:
I don't think that set comprehensions can be included in C407. It works fine in this example, but would be wrong for `sum({x % 2 for x in range(10)})`. As such, the same problem would occur for dict comprehensions.

It may be that the rule needs to be split to functions that are safe for set/dict/list and those that are for list only...

Reply:
You're right. Reverting in #216.

Reply:
Thanks!

> As such, the same problem would occur for dict comprehensions.

Sorry, I probably wasn't clear enough that this is also broken for `dict`:

```python
sum({x % 2: x for x in range(10)})
```

Iteration is over keys which are also set-like.

Reply:
Yes you're right, I should have noticed too. Fixed in #218 and released in 3.2.2.